ocrd/ocrd/workspace.py

```diff
@@ -1024,7 +1025,17 @@ def save_image_file(self, image,
         if not force and self.overwrite_mode:
             force = True
         image_bytes = io.BytesIO()
-        image.save(image_bytes, format=MIME_TO_PIL[mimetype])
+        if dpi and not isinstance(dpi, tuple):
+            dpi = (dpi, dpi)
+        elif not dpi:
+            # TODO - brittle, will this always find the original image?
+            orig_img_file = self.mets.find_files(pageId=page_id, mimetype='//^image')[0]
```

Comment:
We must adapt for the generator now, I believe.

```suggestion
            orig_img_file = next(self.mets.find_files(pageId=page_id, mimetype='//^image', local_only=True))
```